src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,120 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || 'All';
+  const query = searchParams.get('query') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then((peopleFromServer) => setPeople(peopleFromServer))
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const getFilteredPeople = () => {
```

Comment:
Let's move this function to separate file and receive `people` and `filters` as a parameters